### PR TITLE
OL-3

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: web/modules/custom/openideal_challenge/src/Service/OpenidealChallengeServiceInterface.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  /**
+   * Processing for closing / opening scheduled nodes.
+   *
+   * @param string $operation
+   *   The name of operation 'open/close'.
+   *
+   * @return bool
+   *   TRUE if any node has been closed / opened, FALSE otherwise.
+   */
+  public function challengeOpenCloseOperation($operation = 'open');
```
Comment (db-Matroskeen) [inline web/modules/custom/openideal_challenge/src/Service/OpenidealChallengeServiceInterface.php]: I would like to have two **public** methods: openChallenge() and closeChallenge() that will execute **protected** method challengeOpenCloseOperation() with the appropriate parameter.

It'll make this API more intuitive and easier to use for other developers.

Probably, it also should be changed to plural form, as we dealing with challenges (not single challenge).

File: config/sync/core.entity_form_display.node.challenge.default.yml [hunk 2]
```diff
@@ -72,6 +87,17 @@ content:
     region: content
     settings: {  }
     third_party_settings: {  }
+  publish_on:
+    weight: 33
+    settings: {  }
+    third_party_settings: {  }
+    type: datetime_timestamp
+    region: content
+  scheduler_settings:
```
Comment (db-Matroskeen) [inline config/sync/core.entity_form_display.node.challenge.default.yml]: Leftovers from "Scheduler" module? Let's clean-up.

File: web/modules/custom/openideal_challenge/openideal_challenge.module [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+/**
+ * Implements hook_help().
+ */
+function openideal_challenge_help($route_name, RouteMatchInterface $route_match) {
+  switch ($route_name) {
+    // Main module help for the openideal_challenge module.
+    case 'help.page.openideal_challenge':
+      $output = '';
+      $output .= '<h3>' . t('About') . '</h3>';
+      $output .= '<p>' . t('My Awesome Module') . '</p>';
```
Comment (db-Matroskeen) [inline web/modules/custom/openideal_challenge/openideal_challenge.module]: Let's either remove this or provide some useful description :)

File: web/modules/custom/openideal_challenge/src/Service/OpenidealChallengeService.php [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+    if (empty($nids)) {
+      return $result;
+    }
+
+    $nodes = $this->entityTypeManager->getStorage('node')
+      ->loadMultiple($nids);
+    /** @var \Drupal\node\NodeInterface $node */
+    foreach ($nodes as $node) {
+      if ($node->hasField('field_is_open')) {
```
Comment (db-Matroskeen) [inline web/modules/custom/openideal_challenge/src/Service/OpenidealChallengeService.php:100]: Should we move this condition to the query above?
If, for some reason, there is no field "Is Open", then we should probably ignore the whole process.

File: web/modules/custom/openideal_challenge/src/Service/OpenidealChallengeServiceInterface.php (hunk 1 above)
Comment (ghost) [reply web/modules/custom/openideal_challenge/src/Service/OpenidealChallengeServiceInterface.php]: I've changed the code according to your notes, thanks.

File: config/sync/core.entity_form_display.node.challenge.default.yml (hunk 2 above)
Comment (ghost) [reply config/sync/core.entity_form_display.node.challenge.default.yml]: Done, tnx!

File: web/modules/custom/openideal_challenge/openideal_challenge.module (hunk 3 above)
Comment (ghost) [reply web/modules/custom/openideal_challenge/openideal_challenge.module]: Fixed.

**Pull request merged**